source/develop/release-management/features/gluster-self-heal-monitoring.html.md

```diff
@@ -41,21 +41,51 @@ Host Maintenance and Fencing will be enabled/disabled based on the following cri
 | No Unsynced Entries | Quorum available without this Node | Allowed |
 | No Unsynced Entries | Quorum not available without this Node | Not Allowed |
 
+ More info about standard host fencing is available at http://old.ovirt.org/Automatic_Fencing#Automatic_Fencing http://old.ovirt.org/Fence_kdump https://www.youtube.com/watch?v=V1JQtmdleaM
+
 ## Entity Changes
 Following entities will be changed as part of Gluster self heal monitoring.
 
 ### Gluster_volume_bricks
   Following columns will be added to gluster_volume_bricks table.
   unsynced_entries  - integer - No.of unsynced entries in the brick.
   unsynced_entires_history - text - History of unsynced entries in the brick. It will a list of comma separated values.
-  
-### Gluster_volumes_view
-‘heal_status’ column will be be added to gluster_volumes_view. It will be computed based on unsynced_entries in the brick. If there is any unsynced_entry in any of the bricks in the volume then ‘heal_status’ will be ‘Needs_Healing’.
 
+## Change in BLL Commands
+###VdsNotRespondingTreatmentCommand (for Host fencing)
+   VdsNotRespondingTreatmentCommand is the central BLL command which takes care of varios host fencing related activites. This will be changed to consider gluster self-heal information during host fencing. shouldFencingBeSkipped method will be changed and following logic will be added.
+   
+   if (host supports both gluster & virt services)
+      if(all bricks are down)
+        allow host fencing
+      else(some || all bricks are up)
+        if (unsync'd entries  > 0 for any up brick) //Entris needs to be healed from this host
+          skip host fencing
+        else
+          if (quorum is maintained (bricks and nodes for ALL volumes) with this host down)
+            allow host fencing
+          else
+            skip host fencing
+
```

Comment:
Concept looks good, but I have several questions/notes:
1. All of the checks mentioned above are performed using only informations contained in engine db, right? There won't be any remote call to host to get this informations, right?
2. Those gluster related checks should be performed after "fence disable by cluster policy" and "connectivity broken threshold reached". Do you agree?
3. And the most important thing is timing issue: you stated that healing informations are refreshed only each 10 minutes, which is a quite long period and lots of things can happen during that time. How will we handle following scenario:
  a. Host-A is used as a source for healing Host-B and this information is contained in engine db
  b. After a while Host-A crashes due to some kernel or even hardware issue
  c. Engine will start NonRespondingTreatment, but fencing will be skipped due to no-longer-relevant healing informations in engine db -> Host-A will stay in NonResponsive status forever -> HA VM cannot be migrated until host is fenced (which will never happed due to above) or until administator will not mark host as rebooted manually
4. If we add those gluster logic into shouldFencingBeSkipped(), SSH Soft Fencing won't be executed, although during the call we agreed that SSH Soft Fencing cannot harm gluster heal process. So, wouldn't it be better to create different method (for example shouldPmRestartBeSkipped) which would be called prior to PM restart execution?